libminifi/src/io/ClientSocket.cpp

```diff
@@ -207,12 +209,31 @@ int8_t Socket::createConnection(const addrinfo *p, in_addr_t &addr) {
         } else {
           logger_->log_error("Unknown error");
         }
-
-#endif
         closeStream();
         socket_file_descriptor_ = -1;
         return -1;
       }
+#else
+      auto *sa_loc = (struct sockaddr_in*) p->ai_addr;
+      sa_loc->sin_family = AF_INET;
+      sa_loc->sin_port = htons(port_);
+      // use any address if you are connecting to the local machine for testing
+      // otherwise we must use the requested hostname
+      if (IsNullOrEmpty(requested_hostname_) || requested_hostname_ == "localhost") {
+        if (is_loopback_only_) {
+          sa_loc->sin_addr.s_addr = htonl(INADDR_LOOPBACK);
+        } else {
+          sa_loc->sin_addr.s_addr = htonl(INADDR_ANY);
+        }
+      } else {
+        sa_loc->sin_addr.s_addr = addr;
+      }
+      if (connect(socket_file_descriptor_, p->ai_addr, p->ai_addrlen) == -1) {
```

Comment:
I would prefer <= 0 or != 0 checks. In case someone changes the code by introducing a new error this silently gets broken. 

I know it's not completely new code, more like a wish to do it nicely as we touch.

Reply:
I didn't reason about every line of touched code, although I did about some. I wouldn't expect `connect` to ever change its behavior, but I'd like a `<= 0` check better, too.
Notes, will fix.

Reply:
Changed to `< 0` since `0` is a valid file descriptor.